source_mesh.py

```diff
@@ -0,0 +1,381 @@
+
+        # Create tetrahedra for wedges at center of plasma
+        for k, theta in enumerate(theta_list[:-1], 1):
+            # Define indices for wedges at center of plasma
+            wedge_id0 = ma_id
+            wedge_id1 = ma_id + k
+            wedge_id2 = ma_id + k + 1
+            wedge_id3 = next_ma_id
+            wedge_id4 = next_ma_id + k
+            wedge_id5 = next_ma_id + k + 1
+
+            # Define three tetrahedra for wedge
+            create_tet(
+                mbc, tag_handle, tet_set, mbc_verts, verts, verts_s, wedge_id1,
+                wedge_id2, wedge_id4, wedge_id0, strengths
+            )
+            create_tet(
+                mbc, tag_handle, tet_set, mbc_verts, verts, verts_s, wedge_id5,
+                wedge_id4, wedge_id2, wedge_id3, strengths
+            )
+            create_tet(
+                mbc, tag_handle, tet_set, mbc_verts, verts, verts_s, wedge_id0,
+                wedge_id2, wedge_id4, wedge_id3, strengths
+            )
+
+        # Create tetrahedra for hexahedra beyond center of plasma
+        for j, s in enumerate(s_list[1:-1]):
+            # Define index at current closed magnetic flux surface
+            s_offset = j*num_theta
+            # Define index at next closed magnetic flux surface
+            next_s_offset = s_offset + num_theta
+
+            # Create tetrahedra for current hexahedron
+            for k, theta in enumerate(theta_list[:-1], 1):
+                # Define indices for hexahedron beyond center of plasma
+                tet_id0 = ma_id + s_offset + k
+                tet_id1 = ma_id + next_s_offset + k
+                tet_id2 = ma_id + next_s_offset + k + 1
+                tet_id3 = ma_id + s_offset + k + 1
+                tet_id4 = next_ma_id + s_offset + k
+                tet_id5 = next_ma_id + next_s_offset + k
+                tet_id6 = next_ma_id + next_s_offset + k + 1
+                tet_id7 = next_ma_id + s_offset + k + 1
+
+                # Define five tetrahedra for hexahedron
+                create_tet(
+                    mbc, tag_handle, tet_set, mbc_verts, verts, verts_s,
+                    tet_id0, tet_id3, tet_id1, tet_id4, strengths
+                )
+                create_tet(
+                    mbc, tag_handle, tet_set, mbc_verts, verts, verts_s,
+                    tet_id7, tet_id4, tet_id6, tet_id3, strengths
+                )
+                create_tet(
+                    mbc, tag_handle, tet_set, mbc_verts, verts, verts_s,
+                    tet_id2, tet_id1, tet_id3, tet_id6, strengths
+                )
+                create_tet(
+                    mbc, tag_handle, tet_set, mbc_verts, verts, verts_s,
+                    tet_id5, tet_id6, tet_id4, tet_id1, strengths
+                )
+                create_tet(
+                    mbc, tag_handle, tet_set, mbc_verts, verts, verts_s,
+                    tet_id3, tet_id1, tet_id4, tet_id6, strengths
+                )
```

Comment:
create a list and/or numpy array of these canonical orderings and loop over them

Reply:
Do this in a `create_tets_from_hex()` method